std/haxe/macro/Expr.hx

```diff
@@ -296,18 +301,22 @@ typedef Case = {
 typedef Var = {
 	/**
 		The name of the variable.
+		
+		Haxe variable names start with an underscore `_`, a dollar `$`, a lower-case
+		character `a-z` or an upper-case character `A-Z`. After that, any combination 
+		and number of `_`, `A-Z`, `a-z` and `0-9` may follow.
```

Comment:
This information suggests that the naming is somehow enforced. I think the only check that is done after parsing is the leading dollar, so this is somewhat misleading.

It _might_ be useful to mention that this is only guaranteed to work if the name conforms to the Haxe identifier rules.